### PR TITLE
VAULT-32568: Shutdown node when it's not in the raft config

### PR DESCRIPTION
**Pull request by miagilepner**
### Description
This PR has a removed node shutdown if it sees that it's not in the raft configuration. For extra safety's sake, the node must have previously been present in the configuration in order for the node to shutdown.

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this fix needs to be backported, use the appropriate `backport/` label that matches the desired release branch. Note that in the CE repo, the latest release branch will look like `backport/x.x.x`, but older release branches will be `backport/ent/x.x.x+ent`.
    - [ ] **LTS**: If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [x] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: physical/raft/raft.go [hunk 1]
```diff
@@ -1391,6 +1402,9 @@ func (b *RaftBackend) SetupCluster(ctx context.Context, opts SetupOpts) error {
 		}
 	}
 
+	if opts.RemovedCallback != nil {
+		b.removedCallback = opts.RemovedCallback
```
Comment (kubawi) [inline physical/raft/raft.go:1406]: Should this be a call to `SetRemovedCallback` instead? (Genuine question, I am not sure if we care about locking at this point.)

File: physical/raft/raft.go [hunk 2]
```diff
@@ -1030,6 +1031,12 @@ func (b *RaftBackend) SetRestoreCallback(restoreCb restoreCallback) {
 	b.fsm.l.Unlock()
 }
 
+func (b *RaftBackend) SetRemovedCallback(cb func()) {
```
Comment (kubawi) [inline physical/raft/raft.go:1034]:
```suggestion
func (b *RaftBackend) setRemovedCallback(cb func()) {
```

Can we leave this function unexported?

**Review by kubawi — APPROVED** [review]
LGTM

File: physical/raft/raft.go (hunk 1 above)
Comment (miagilepner) [reply physical/raft/raft.go:1406]: the lock is already grabbed earlier in the function, so we shouldn't grab it again

File: physical/raft/raft.go (hunk 2 above)
Comment (kubawi) [reply physical/raft/raft.go:1034]: We will still be able to access it in `physical/raft/raft_test.go` if it's unexported, unless you mean for future tests outside this PR? Either way no biggie, feel free to resolve the thread 😄

File: physical/raft/raft.go (hunk 1 above)
Comment (kubawi) [reply physical/raft/raft.go:1406]: Yep, missed that 🤦

File: physical/raft/raft.go (hunk 2 above)
Comment (miagilepner) [reply physical/raft/raft.go:1034]: True, I was thinking for any future tests

**Review by raskchanky — APPROVED** [review]

**Pull request merged**